derive_builder_core/src/initializer.rs

```diff
@@ -155,54 +166,66 @@ pub enum FieldConversion<'a> {
 /// To be used inside of `#struct_field: match self.#builder_field { ... }`
 enum MatchNone<'a> {
     /// Inner value must be a valid Rust expression
-    DefaultTo(&'a DefaultExpression),
+    DefaultTo {
+        expr: &'a DefaultExpression,
+        crate_root: &'a syn::Path,
+    },
     /// Inner value must be the field identifier
     ///
     /// The default struct must be in scope in the build_method.
     UseDefaultStructField(&'a syn::Ident),
     /// Inner value must be the field name
-    ReturnError(String, Option<Span>),
+    ReturnError {
+        crate_root: &'a syn::Path,
+        field_name: String,
+        span: Option<Span>,
+    },
 }
 
 impl<'a> ToTokens for MatchNone<'a> {
     fn to_tokens(&self, tokens: &mut TokenStream) {
         match *self {
-            MatchNone::DefaultTo(expr) => tokens.append_all(quote!(
-                None => #expr
-            )),
+            MatchNone::DefaultTo { expr, crate_root } => {
+                let expr = expr.with_crate_root(crate_root);
+                tokens.append_all(quote!(None => #expr));
+            }
             MatchNone::UseDefaultStructField(field_ident) => {
                 let struct_ident = syn::Ident::new(DEFAULT_STRUCT_NAME, Span::call_site());
                 tokens.append_all(quote!(
                     None => #struct_ident.#field_ident
                 ))
             }
-            MatchNone::ReturnError(ref field_name, ref span) => {
+            MatchNone::ReturnError {
+                ref field_name,
+                ref span,
+                crate_root,
+            } => {
                 let conv_span = span.unwrap_or_else(Span::call_site);
-                let err_conv = quote_spanned!(conv_span => ::derive_builder::export::core::convert::Into::into(
-                    ::derive_builder::UninitializedFieldError::from(#field_name)
+                let err_conv = quote_spanned!(conv_span => #crate_root::export::core::convert::Into::into(
+                    #crate_root::UninitializedFieldError::from(#field_name)
```

Comment:
I'm not sure why yet, but this seems to be the problem. Changing `#crate_root` back to a hardcoded value fixes the error span that's failing the compiletest.